HRHN/View/VC/ReviewViewController.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  ReviewViewController.swift
+//  HRHN
+//
+//  Created by 민채호 on 2022/12/26.
+//
+
+import UIKit
+import SnapKit
+
+final class ReviewViewController: UIViewController {
+    private let ReviewViewHC = UIHostingController(rootView: ReviewView())
+    private lazy var UIReviewView = ReviewViewHC.view ?? UIView()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setNavigationBar()
+        setUI()
+        setLayout()
+    }
+    
+    private func setNavigationBar() {
+        navigationController?.navigationBar.topItem?.title = ""
+    }
+    
+    private func setUI() {
+        view.backgroundColor = .background
+    }
+    
+    private func setLayout() {
+        view.addSubview(UIReviewView)
+        UIReviewView.snp.makeConstraints {
+            $0.edges.equalTo(view.safeAreaLayoutGuide)
+        }
+    }
+}
```

Comment:
아무래도 언래핑하는건 안좋으니깐 `addChild` 를 이용해서 이렇게 하는 방법은 어떨까요?

- 참고하면 좋을 것: [SwiftUI and UIKit interoperability - Part 2](https://www.swiftbysundell.com/articles/swiftui-and-uikit-interoperability-part-2/)
  - 화면전체가 아니라 컴포넌트인경우 레퍼런스처럼 직접 레이아웃을 잡아도 좋겠네요!

```suggestion
    private let reviewView = UIHostingController(rootView: ReviewView())
    
    override func viewDidLoad() {
        super.viewDidLoad()
        setNavigationBar()
        setUI()
        setLayout()
    }
    
    private func setNavigationBar() {
        navigationController?.navigationBar.topItem?.title = ""
    }
    
    private func setUI() {
        view.backgroundColor = .background
        // ✨
        addChild(reviewView)
        reviewView.view.frame = self.view.frame
        view.addSubview(reviewView.view)
        reviewView.didMove(toParent: self)
        // ✨
    }
    
```


<img src="https://user-images.githubusercontent.com/63157395/209666501-867811f8-63ec-4a78-ba6c-cc9f58ce61f8.png" width=""300>

Reply:
위 코드를 보니까 그냥 변수로 저장해놓지 않으면 옵셔널 언래핑이 필요가 없네요??
```swift
    private func setLayout() {
        view.addSubview(reviewViewHC.view)
        reviewViewHC.view.snp.makeConstraints {
            $0.edges.equalTo(view.safeAreaLayoutGuide)
        }
    }
```
이렇게 해도 되지 않을까요?

그리고 저 코드를 찾았었는데 이해를 못해서 넘긴것도 있는데 저게 어떤 동작일까요?
addChild랑 didMove를 몰라서,,,

Reply:
https://ios-development.tistory.com/228 뷰컨에서 뷰컨을 쓰려고 할 때 필요한 애들이네요 HostingController는 결국 SwiftUI 기반이라 생명주기가 달라서 이거 안써도 될거같기도 하고... 잘 모르겠네요

Reply:
71db382
뷰컨에 뷰컨을 추가하는데는 addChild가 정석인거 같아 addChild로 HostingController를 추가하고, 레이아웃은 frame을 같게 하는 대신 원래 하던대로 snapKit으로 잡아줬습니다.
<img width="847" alt="image" src="https://user-images.githubusercontent.com/75792767/209670280-80213a1e-edba-46d2-81a5-3c274e34e510.png">
https://www.swiftbysundell.com/articles/swiftui-and-uikit-interoperability-part-2/의 방식을 참고함

Reply:
> 위 코드를 보니까 그냥 변수로 저장해놓지 않으면 옵셔널 언래핑이 필요가 없네요??
> 
> ```swift
>     private func setLayout() {
>         view.addSubview(reviewViewHC.view)
>         reviewViewHC.view.snp.makeConstraints {
>             $0.edges.equalTo(view.safeAreaLayoutGuide)
>         }
>     }
> ```
> 
> 이렇게 해도 되지 않을까요?
> 
> 그리고 저 코드를 찾았었는데 이해를 못해서 넘긴것도 있는데 저게 어떤 동작일까요? addChild랑 didMove를 몰라서,,,

음🤔 저는 UIHostingController 가 UIViewController 의 하위클래스라서 저렇게 하는게 자연스럽다구 생각해서 추천해보았습니다! 
근데 위 방식으로 가능하다면 그방법도 괜찮을 수 있겠네요! 
다른 사람들이 안쓴데는 이유가 있을것같긴한데 스유+유킷 같이 쓰는 거 어렵네용...

Reply:
> > 위 코드를 보니까 그냥 변수로 저장해놓지 않으면 옵셔널 언래핑이 필요가 없네요??
> > ```swift
> >     private func setLayout() {
> >         view.addSubview(reviewViewHC.view)
> >         reviewViewHC.view.snp.makeConstraints {
> >             $0.edges.equalTo(view.safeAreaLayoutGuide)
> >         }
> >     }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 이렇게 해도 되지 않을까요?
> > 그리고 저 코드를 찾았었는데 이해를 못해서 넘긴것도 있는데 저게 어떤 동작일까요? addChild랑 didMove를 몰라서,,,
> 
> 음🤔 저는 UIHostingController 가 UIViewController 의 하위클래스라서 저렇게 하는게 자연스럽다구 생각해서 추천해보았습니다! 근데 위 방식으로 가능하다면 그방법도 괜찮을 수 있겠네요! 다른 사람들이 안쓴데는 이유가 있을것같긴한데 스유+유킷 같이 쓰는 거 어렵네용...

그 위에 댓글전에 달았는데 밀려서 여기까지 오다니! 어프로브했어요!